facefusion/uis/components/webcam.py

```diff
@@ -27,15 +27,17 @@
 WEBCAM_START_BUTTON : Optional[gradio.Button] = None
 WEBCAM_STOP_BUTTON : Optional[gradio.Button] = None
 
-
-def get_webcam_capture() -> Optional[cv2.VideoCapture]:
+def get_webcam_capture(webcam_mode: WebcamMode = None) -> Optional[cv2.VideoCapture]:
```

Comment:
I suggest this:

```
def get_webcam_capture(webcam_mode : WebcamMode, webcam_device : int, webcam_url : str) -> Optional[cv2.VideoCapture]:```

setting None as default is kinda pointless as None is the default when not being set :D

Reply:
I set it to None to make it optional, otherwise the function call will expect an arguemnt and return error, since it had none initially.

Reply:
I can smell an indirect request for a 2 in 1 while we at it; meaning: also get available system cameras and pass the index as webcam device from the list.. huum I'll try to do both in that case, will get back to it later this evening. (EST time of course)

Reply:
Not sure if this makes sense but what about two methods? shared logic can be extracted and reused.

get_local_webcam_capture() and get_remote_webcam_capture() 

We try not to use defaults for arguments - there are just 2 or 3 cases in the whole project. Why? Because it makes it very hard to find bugs when defaults are hidden in the method arguments.